src/simplicial-hott/03-extension-types.rzk.md

```diff
@@ -15,6 +15,105 @@ This is a literate `rzk` file:
 - the file `hott/4-equivalences.rzk` relies in turn on the previous files in
   `hott/`
 
+## Extension up to homotopy
+
+For a shape inclusion `ϕ ⊂ ψ` and any type `A`,
+we have the inbuilt extension types `(t : ψ) → A [ϕ t ↦ σ t]`
+(for every `σ : ϕ → A`).
+
+We show that these extension types are equivalent to the fibers
+of the canonical restriction map `(ψ → A) → (ϕ → A)`,
+which we can view as the types  of "extension up to homotopy".
+
+```rzk
+#section extensions-up-to-homotopy
+#variable I : CUBE
+#variable ψ : I → TOPE
+#variable ϕ : ψ → TOPE
+#variable A : U
+
+#def extension-type
+  ( σ : ϕ → A)
+  : U
+  :=
+    ( t : ψ) → A [ϕ t ↦ σ t]
+
+#def homotopy-extension-type
+  ( σ : ϕ → A)
+  : U
+  := fib (ψ → A) (ϕ → A) (\ τ t → τ t) (σ)
+
+#def extension-type-weakening-map
```

Comment:
Any thoughts about the names? Something like "homotopy-extension-type-extension-type" is more straightforward but perhaps less evocative.